skyportal/handlers/api/spectrum.py

```diff
@@ -174,34 +155,20 @@ def get(self, spectrum_id):
                 schema: Error
         """
 
-        spectrum = (
-            DBSession()
-            .query(Spectrum)
-            .join(Obj)
-            .join(GroupSpectrum)
-            .filter(
-                Spectrum.id == spectrum_id,
-                GroupSpectrum.group_id.in_(
-                    [g.id for g in self.current_user.accessible_groups]
-                ),
-            )
-            .options(joinedload(Spectrum.groups))
-            .first()
+        query_options = [
+            joinedload(Spectrum.instrument.name)
+            .joinedload(Spectrum.groups)
+            .joinedload(Spectrum.reducers)
+            .joinedload(Spectrum.observers)
+            .joinedload(Spectrum.owner)
+        ]
+        spectrum = Spectrum.get_if_accessible_by(
+            spectrum_id, self.current_user, raise_if_none=True, options=query_options
         )
 
-        if spectrum is not None:
-            # Permissions check
-            _ = Obj.get_if_readable_by(spectrum.obj_id, self.current_user)
-            spec_dict = spectrum.to_dict()
-            spec_dict["instrument_name"] = spectrum.instrument.name
-            spec_dict["groups"] = spectrum.groups
-            spec_dict["reducers"] = spectrum.reducers
-            spec_dict["observers"] = spectrum.observers
-            spec_dict["owner"] = spectrum.owner
-            self.verify_and_commit()
-            return self.success(data=spec_dict)
-        else:
-            return self.error(f"Could not load spectrum with ID {spectrum_id}")
+        spec_dict = spectrum.to_dict()
```

Comment:
You'll need to use `baselayer`'s `recursive_to_dict` here in order to preserve the loaded relationships (see current source or candidate handler for examples)

Reply:
Updated with latest commit